### PR TITLE
MM-48048 Update HTML lang attribute when a user updates locale

### LINKED ISSUE #21552: The HTML tag always has `lang="en"` even if the language in settings is set to a different language
#### Summary
Title. There is always `<html lang="en">` even if the settings specify a different language. This matters because I have joined a German server and don't speak any German. I want the `lang` attribute to match the user-generated messages so that Chrome will translate the entire page. Setting the language in settings to English does not solve this because the German messages will not be translated.

#### Steps to reproduce
1. Join a German server or change the language to German in Settings > Display > Change interface language
2. Notice that the HTML tag still has `lang="en"`

#### Expected behavior
The `lang` attribute on the HTML tag matches the configured interface language.

#### Observed behavior (that appears unintentional) 
The `lang` attribute on the HTML tag always has the value "en"

### PR DESCRIPTION
**Pull request by gitstart**
### Summary
The HTML tag always has lang="en" even if the language in settings is set to a different language

#### Demo video
https://www.loom.com/share/19060c7452614ae984e31bc3fb757223

- Server PR https://github.com/mattermost/mattermost-server/pull/22262

#### Release note
```release-note
Updates HTML lang attribute when a user updates locale
```

---
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

File: components/root/root_provider.tsx [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
 
 import {WebSocketContext} from 'utils/use_websocket';
+import {getCurrentLocale} from 'selectors/i18n';
 
 type Props = {
     children: React.ReactNode;
 }
 
 export default function RootProvider(props: Props) {
+    const locale = useSelector(getCurrentLocale);
```
Comment (M-ZubairAhmed) [inline components/root/root_provider.tsx:20]: better place for this could be `components/root/root.tsx` instead here

**Comment by gitstart** [comment]
@hmhealey You can have a look. Thanks.

**Review by hmhealey — CHANGES_REQUESTED** [review]
This looks pretty good to me, but could we actually move this logic into a subcomponent into a different component to avoid adding more logic to the `Root` component? The `IntlProvider` component `components/intl_provider` already deals with some stuff around the locale, so I think it would make sense to have it there.

Could you also add a unit test to that component to ensure that the lang is updated correctly when `currentLocale` changes?

**Comment by collinewait** [comment]
> This looks pretty good to me, but could we actually move this logic into a subcomponent into a different component to avoid adding more logic to the `Root` component? The `IntlProvider` component `components/intl_provider` already deals with some stuff around the locale, so I think it would make sense to have it there.
> 
> Could you also add a unit test to that component to ensure that the lang is updated correctly when `currentLocale` changes?

@hmhealey This has been addressed. Thanks.

**Review by hmhealey — CHANGES_REQUESTED** [review]
Thanks for making that change! That looks perfect to me.

One last thing though, could we actually remove the `lang="en"` from `root.html`? I think it's better to not set it at all until the `IntlProvider` is loaded

**Comment by gitstart** [comment]
@hmhealey root.html has been updated.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]
Thanks! This looks good to me

**Comment by stevemudie** [comment]
@hmhealey The E2E won't run and the Spinwick won't build either so I'm not able to complete tests on this.

As a guess it has something to do with the failed check listed below?

**Comment by hmhealey** [comment]
I'll try updating this to the latest version of master again. I assume that there's something here broken by the new CI process

/update-branch

**Comment by M-ZubairAhmed** [comment]
Hey there!, Thank you for your PR.
Webapp is now part of mattermost monorepo now at https://github.com/mattermost/mattermost-server/tree/master/webapp Would you be interested in opening the PR over there?

**Comment by gitstart** [comment]
@hmhealey This PR has been moved to the monorepo https://github.com/mattermost/mattermost-server/pull/22677. cc @M-ZubairAhmed

**Pull request closed without merging**